### PR TITLE
Make code functional

### PR DESCRIPTION
**Pull request by ken0nek**
- use `reduce` not `for-loop`

### CODE REVIEW

File: ObjectMapper/Core/Mapper.swift [hunk 1]
```diff
@@ -412,38 +412,31 @@ extension Mapper where N: Hashable {
 }
 
 extension Dictionary {
-	internal func map<K: Hashable, V>(@noescape f: Element -> (K, V)) -> [K : V] {
```
Comment (chainone) [inline ObjectMapper/Core/Mapper.swift]: For such a low layer of abstraction, I don't think applying "reduce" everywhere is a good idea. Performance hurts.

Comment (chainone) [inline ObjectMapper/Core/Mapper.swift]: Let me know if I am wrong.

Comment (tristanhimmelman) [inline ObjectMapper/Core/Mapper.swift]: I have not measured the performance of reduce so I don't know if this is true. However I don't see a compelling reason to use it in this situation. I personally find the syntax is a bit clearer without it.

Comment (chainone) [inline ObjectMapper/Core/Mapper.swift]: Take look at this blog:
http://airspeedvelocity.net/

"Your reminder that building arrays with reduce, while fun, is accidentally quadratic."

**Comment by tristanhimmelman** [comment]
Thanks for the help but I've decided not to include this PR based on the above conversation.

**Pull request closed without merging**